src/coreclr/jit/lower.cpp

```diff
@@ -8195,6 +8198,10 @@ bool Lowering::TryLowerConstIntDivOrMod(GenTree* node, GenTree** nextNode)
 #elif defined(TARGET_ARM)
         // Currently there's no GT_MULHI for ARM32
         return false;
+#elif defined(TARGET_WASM)
+        // TODO-WASM: it is not clear this whole transformation is profitable on WASM, since it increases
+        // code size while the WASM VM should be perfectly capable of expanding the DIV/MOD by itself.
+        NYI_WASM("Lowering::TryLowerConstIntDivOrMod");
```

Comment:
Good call, I agree with your reasoning. We might want to just return false here, but the first person to hit the NYI can probably do that instead of us needing to do it now.